Realm/Realm/Handles/SharedRealmHandle.cs

```diff
@@ -523,9 +523,20 @@ public IntPtr ResolveReference(ThreadSafeReference reference)
             return result;
         }
 
-        public void WriteCopy(string path, byte[] encryptionKey)
+        public void WriteCopy(RealmConfigurationBase config)
         {
-            NativeMethods.write_copy(this, path, (IntPtr)path.Length, encryptionKey, out var nativeException);
+            IntPtr syncConfiguration;
+            if (config is SyncConfigurationBase syncConfig)
+            {
+                var configHandle = GCHandle.Alloc(syncConfig.CreateNativeSyncConfiguration());
+                syncConfiguration = GCHandle.ToIntPtr(configHandle);
```

Comment:
This seems wrong - we should just pass the sync config as a value similarly to what we're doing in `OpenWithSync`.